features/admin.actions.v1/components/action-config-form.tsx

```diff
@@ -0,0 +1,829 @@
+                        error.valueAuthProperty = t("console:manage.features.actions.fields.authentication." +
+                            "types.apiKey.properties.value.validations.empty");
+                    }
+                }
+
+                break;
+            default:
+                break;
+        }
+
+        return error;
+    };
+
+    const handleSubmit = async (values: ActionConfigFormPropertyInterface) => {
+        const authProperties: Partial<AuthenticationPropertiesInterface> = {};
+
+        if (isAuthenticationUpdateFormState || isCreateFormState) {
+            switch (authenticationType) {
+                case AuthenticationType.BASIC:
+                    authProperties.username = values.usernameAuthProperty;
+                    authProperties.password = values.passwordAuthProperty;
+
+                    break;
+                case AuthenticationType.BEARER:
+                    authProperties.accessToken = values.accessTokenAuthProperty;
+
+                    break;
+                case AuthenticationType.API_KEY:
+                    authProperties.header = values.headerAuthProperty;
+                    authProperties.value = values.valueAuthProperty;
+
+                    break;
+                case AuthenticationType.NONE:
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (isCreateFormState) {
+            const actionValues: ActionInterface = {
+                endpoint: {
+                    authentication: {
+                        properties: authProperties,
+                        type: authenticationType
+                    },
+                    uri: values.endpointUri
+                },
+                name: values.name
+            };
+
+            setIsSubmitting(true);
+            createAction(actionTypeApiPath, actionValues)
+                .then(() => {
+                    handleSuccess(ActionsConstants.CREATE);
+                    mutateActions();
+                })
+                .catch((error: AxiosError) => {
+                    handleError(error, ActionsConstants.CREATE);
+                })
+                .finally(() => {
+                    setIsSubmitting(false);
+                });
+        } else {
+            // Updating the action
+            const updatingValues: ActionUpdateInterface = {
+                endpoint: isAuthenticationUpdateFormState || values.endpointUri !== initialValues?.endpointUri ? {
+                    authentication: isAuthenticationUpdateFormState ? {
+                        properties: authProperties,
+                        type: authenticationType
+                    } : undefined,
+                    uri: values.endpointUri !== initialValues?.endpointUri ? values.endpointUri : undefined
+                } : undefined,
+                name: values.name !== initialValues?.name ? values.name : undefined
+            };
```

Comment:
1. Any possibility of `isAuthenticationUpdateFormState` being `false` here? AFAIU it'll never be `false` inside this else block.
2. Checking the changed status of fields can be improved by using FormAPI itself. Validate whether this breaks with conditional fields.
Ex: Pass the `form` prop to the callback and access changed fields using that.
```
<FinalForm
            onSubmit={ (values, form) => {
                console.log(form.getState().dirtyFields);

                handleSubmit(values, form);
            } }
```

Reply:
When updating we are giving the capability to update fields separately. Which means we can update the action without the authentication section. In such scenarios we are not sending authentication information to the back end because updating authentication involves few extra steps including secret encryption.

Reply:
used dirty fields to check changed fields.
Fixed with https://github.com/wso2/identity-apps/pull/6722/commits/5e3564e0797c09f4f20b71837b928bc9fa57c847